src/main/webapp/src/app/models/geometry.ts

```diff
@@ -0,0 +1,9 @@
+import { Location } from './location';
+
+export interface Geometry {
+  location: Location;
```

Comment:
Could you add some comments describing what these are?
I can guess what location is but I'm not familiar with viewport.
Also meta question do you need viewport? you could save your service some bandwidth by avoiding pushing up data you won't use. (though this might be what sehhaj is about to do, if so don't worry)

Reply:
I have kept viewport for now but, yes, Sehhaj is currently working on the Hidden Gem object to avoid pushing these unused data.